### PR TITLE
[CONTP-222] Autodiscovery: fix controller register and unit test

### PR DESCRIPTION
**Pull request by zhuminyi**
### What does this PR do?
[CONTP-222](https://datadoghq.atlassian.net/browse/CONTP-222)
1. Fix flaky unit test 
2. Fix double scheduling config in register function (by reverting back to old logic)

### Motivation

Follow up recent [PR](https://github.com/DataDog/datadog-agent/pull/24350) for autodiscovery controller
1. Flaky unit test was caused by the failure to inject mock clock to controller. New test is using assert.EventuallyWithTf and assert.Never to check configs are properly scheduled. 
2. During the unit test,  one potential bug in register function was discovered when replayConfigs is true. When we replay the currently-scheduled configs, the schedule call must be protected by lock. Otherwise, there is a race condition competing from worker [function](https://github.com/DataDog/datadog-agent/blob/539f7e9aa54c29289b3fe849db1bb9aebfecda3c/comp/core/autodiscovery/scheduler/controller.go#L154) which caused double scheduling in certain case. Note: The change in this PR is reverting register function to original logic (see original workflow [comment](https://github.com/DataDog/datadog-agent/blob/7.54.x/comp/core/autodiscovery/scheduler/meta.go#L50))

### Describe how to test/QA your changes
Skip because
1. Unit test fix is verified in CI
2. Register function is reverted back, nothing new. 

[CONTP-222]: https://datadoghq.atlassian.net/browse/CONTP-222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: comp/core/autodiscovery/scheduler/controller_test.go [hunk 1]
```diff
@@ -70,53 +65,62 @@ func TestController(t *testing.T) {
...
 	s1 := &scheduler{}
 	ms.Register("s1", s1, true)
-	s1.mutex.Lock()
-	require.ElementsMatch(t, []event{{true, "one"}, {true, "two"}}, s1.events)
-	s1.mutex.Unlock()
-
-	ms.ApplyChanges(integration.ConfigChanges{Schedule: []integration.Config{c1, c2}})
-	clk := clock.NewMock()
-	clk.Add(2000 * time.Millisecond)
+	time.Sleep(2 * time.Second)
```
Comment (clamoriniere) [inline comp/core/autodiscovery/scheduler/controller_test.go:68]: do we need this `sleep()` since after we have a `assert.EventuallyWithTf`

Comment (zhuminyi) [reply comp/core/autodiscovery/scheduler/controller_test.go:68]: Yes, it is for checking if there is any config double scheduled.

**Review by wdhif — APPROVED** [review]

**Pull request merged**